### ISSUE TITLE
me saved issue with comments

### ISSUE THREAD

**Issue by vikigenius** [halcyonnouveau/roux#36]
So it seems like me saved is supposed to include comments, but the target type is just submissionsdata, which causes errors such as:

When I do:
```rs
    let saved = me.saved().await.unwrap();
    println!("{}", saved.data.children[0].data.selftext);
```

I get the following error (with backtrace) probably because my first saved item is a comment.

```rs
---- core::my_test stdout ----
thread 'core::my_test' panicked at 'called `Result::unwrap()` on an `Err` value: Network(reqwest::Error { kind: Decode, source: Error("missing field `domain`", line: 1, column: 4016) })', src/core/mod.rs:8:34
stack backtrace:
   0: rust_begin_unwind
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:517:5
   1: core::panicking::panic_fmt
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/panicking.rs:100:14
   2: core::result::unwrap_failed
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/result.rs:1616:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/result.rs:1298:23
   4: redorg::core::my_test::{{closure}}
             at ./src/core/mod.rs:8:17
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/future/mod.rs:80:19
   6: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/future/future.rs:119:9
   7: tokio::runtime::basic_scheduler::Inner<P>::block_on::{{closure}}::{{closure}}
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:240:62
   8: tokio::coop::with_budget::{{closure}}
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:102:9
   9: std::thread::local::LocalKey<T>::try_with
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/thread/local.rs:399:16
  10: std::thread::local::LocalKey<T>::with
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/thread/local.rs:375:9
  11: tokio::coop::with_budget
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:95:5
  12: tokio::coop::budget
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:72:5
  13: tokio::runtime::basic_scheduler::Inner<P>::block_on::{{closure}}
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:240:39
  14: tokio::runtime::basic_scheduler::enter::{{closure}}
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:349:29
  15: tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/macros/scoped_tls.rs:61:9
  16: tokio::runtime::basic_scheduler::enter
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:349:5
  17: tokio::runtime::basic_scheduler::Inner<P>::block_on
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:230:9
  18: tokio::runtime::basic_scheduler::InnerGuard<P>::block_on
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:501:9
  19: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/basic_scheduler.rs:186:24
  20: tokio::runtime::Runtime::block_on
             at /home/void/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/mod.rs:461:46
  21: redorg::core::my_test
             at ./src/core/mod.rs:9:5
  22: redorg::core::my_test::{{closure}}
             at ./src/core/mod.rs:2:7
  23: core::ops::function::FnOnce::call_once
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/ops/function.rs:227:5
  24: core::ops::function::FnOnce::call_once
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/ops/function.rs:227:5
```

**Comment by beanpuppy** [comment]
Thanks for the report, I've changed the return type of saved to an enum with comments and submissions here: [src/me/responses/me.rs](https://github.com/halcyonnouveau/roux/commit/aecefbdf8429acd2cb486348be74e462cf9a8a2d#diff-3cfeed0fa7e45ec7f5337b3ace85f37b8156b6f4d565017270ebde690001d9a7).

**Comment by vikigenius** [comment]
@beanpuppy thanks for fixing this. I do have a question about how to actually iterate through the saved items or any listings in general. You seem to directly return the BasicListing, are we supposed to be provide the iterators ourselves and repeatedly make use of `after` and `children` ?

**Comment by beanpuppy** [comment]
Yes, you should be iterating them using `after` and the `FeedOption` util, but it looks like I forgot to add `FeedOption` to some endpoints for `me`. This has been fixed here: c811b15

So now iterating should look something like this:

```rust
use roux::util::FeedOption;

// Get first page saved
let saved = me.saved(None).await.unwrap();
// Create options with `after` from saved
let after_options = FeedOption::new().after(&saved.data.after.unwrap());
// Get next page saved
let next_saved = me.saved(Some(after_options)).await.unwrap();
```
